service.py

```diff
@@ -315,9 +315,9 @@ def get_subject(subjectID):
     ).all()
     for x in a:
         hot_article.append(x)
-
-    return render_template('subject.html', url=url, subject_id=subject.id, articles=articles, hot_article=hot_article, Tool=Tool)
-
+    if not subject.pid  == "None" :
```

Comment:
@mmmmmkj It is strange that you make None a string: **"**None**"**.  Are you sure it is correct?

-Hui